testplan/common/utils/monitor/collectors.py

```diff
@@ -0,0 +1,121 @@
+            try:
+                counters = {
+                    'io_counter': raw_data['io_counters'].read_count + raw_data['io_counters'].write_count,
+                    'read_counter': raw_data['io_counters'].read_bytes,
+                    'write_counter': raw_data['io_counters'].write_bytes,
+                    'time': time.time()
+                }
+                iops = read = write = 0
+                if raw_data['pid'] in self._io_counter \
+                        and self._io_counter[raw_data['pid']]['io_counter'] >= counters['io_counter']:
+                    _interval = counters['time'] - self._io_counter[raw_data['pid']]['time']
+                    iops = (counters['io_counter'] - self._io_counter[raw_data['pid']]['io_counter']) / _interval
+                    read = (counters['read_counter'] - self._io_counter[raw_data['pid']]['read_counter']) / _interval
+                    write = (counters['write_counter'] - self._io_counter[raw_data['pid']]['write_counter']) / _interval
+                else:
+                    _interval = counters['time'] - raw_data['create_time']
+                    if _interval >= 1.0:
+                        iops = counters['io_counter'] / _interval
+                        read = counters['read_counter'] / _interval
+                        write = counters['write_counter'] / _interval
+                self._io_counter[raw_data['pid']] = counters.copy()
+                monitor_result['iops'] += iops
+                monitor_result['read'] += read
+                monitor_result['write'] += write
+            except (AttributeError, KeyError):
+                # process exited or no io counters
+                pass
+        return monitor_result
+
+
+class LinuxCollector(Collector):
+    def __init__(self, director):
+        super(LinuxCollector, self).__init__(director)
+        try:
+            cpu_info = subprocess.check_output(['lscpu'])
+            self.metadata['extra']['cpu_info'] = cpu_info.decode()
+        except Exception:
+            pass
+
+
+class LinuxContainerCollector(LinuxCollector):
+    def __init__(self, director):
+        super(LinuxContainerCollector, self).__init__(director)
+        with open('/proc/1/cgroup') as cgroup_file:
+            for cgroup in cgroup_file:
+                try:
+                    if ':memory:' in cgroup:
+                        path = os.path.join('/sys/fs/cgroup/memory/', cgroup.split(':')[-1][1:],
+                                            'memory.limit_in_bytes')
+                        with open(path, 'r') as memory_file:
+                            self.metadata['memory'] = int(memory_file.read())
+                except (IOError, AttributeError, ValueError):
+                    pass
+
+
+class WindowsCollector(Collector):
+    def __init__(self, director):
+        super(WindowsCollector, self).__init__(director)
+
+
+def get_collector(director=os.getcwd()):
```

Comment:
we use an internal utility instead of os.getcwd()
https://github.com/Morgan-Stanley/testplan/blob/master/testplan/common/utils/path.py#L36